core/domain/collection_services.py

```diff
@@ -1020,14 +1020,57 @@ def compute_summary_of_collection(collection, contributor_id_to_add):
         collection.objective, collection.language_code, collection.tags,
         collection_rights.status, collection_rights.community_owned,
         collection_rights.owner_ids, collection_rights.editor_ids,
-        collection_rights.viewer_ids, contributor_ids, contributors_summary,
+        collection_rights.viewer_ids, collection_summary_model.contributor_ids,
+        collection_summary_model.contributors_summary,
         collection.version, collection_model_node_count,
         collection_model_created_on, collection_model_last_updated
     )
 
     return collection_summary
 
 
+def add_new_contributor_to_collection_summary(
```

Comment:
Optional: Should we make this as CollectionSummary object function? [Considering this entire method is object dependent.]

Reply:
Done.